### PR TITLE
feat(linter): linter-eslint-plugin-import/no-named-as-default

### PR DESCRIPTION
**Pull request by XantreDev**
#1117 

Code: https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-named-as-default.js 
Tests: https://github.com/import-js/eslint-plugin-import/blob/main/tests/src/rules/no-named-as-default.js
Docs: https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-named-as-default.md

### CODE REVIEW

File: crates/oxc_linter/src/snapshots/no_named_as_default.snap [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+source: crates/oxc_linter/src/tester.rs
+assertion_line: 143
+expression: no_named_as_default
+---
+  ⚠ eslint-plugin-import(no-named-as-default): Using exported name "foo" as identifier for default export.
```
Comment (Boshen) [inline crates/oxc_linter/src/snapshots/no_named_as_default.snap:6]: Do we need to say what the export name is from the other side? The current messages doesn't say about why this is bad.

Comment (Boshen) [inline crates/oxc_linter/src/snapshots/no_named_as_default.snap:6]: Do we need to say what the export name is from the other side? The current messages doesn't say about why this is bad.

Comment (XantreGodlike) [reply crates/oxc_linter/src/snapshots/no_named_as_default.snap:6]: I've used [same](https://github.com/import-js/eslint-plugin-import/blob/7a21f7e10f18c04473faadca94928af6b8e28009/tests/src/rules/no-named-as-default.js#L71C1-L72C1) error message as in eslint plugin. What do you want to add in it?

Comment (XantreGodlike) [reply crates/oxc_linter/src/snapshots/no_named_as_default.snap:6]: Maybe something like: `Module './bar' has named export 'foo'. Naming default import as 'foo' can be confusing`

Comment (Boshen) [reply crates/oxc_linter/src/snapshots/no_named_as_default.snap:6]: @XantreGodlike sounds good to me.

Comment (XantreGodlike) [reply crates/oxc_linter/src/snapshots/no_named_as_default.snap]: @Boshen fixed

**Comment by Boshen** [comment]
Thank you! What a confusing rule 😅

**Pull request merged**